### ISSUE TITLE
throwIndexOutOfBoundsException

### ISSUE THREAD

**Issue by ckenergy** [PhilJay/MPAndroidChart#3696]
```
12-28 15:07:06.506 25737-25737/com.xxmassdeveloper.mpchartexample E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                    Process: com.xxmassdeveloper.mpchartexample, PID: 25737
                                                                                    java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
                                                                                        at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
                                                                                        at java.util.ArrayList.get(ArrayList.java:308)
                                                                                        at com.github.mikephil.charting.data.DataSet.getEntryForIndex(DataSet.java:286)
                                                                                        at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesLine(Transformer.java:184)
                                                                                        at com.github.mikephil.charting.renderer.LineChartRenderer.drawValues(LineChartRenderer.java:547)
                                                                                        at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:264)
                                                                                        at android.view.View.draw(View.java:14768)
                                                                                        at android.view.View.getDisplayList(View.java:13615)
                                                                                        at android.view.View.getDisplayList(View.java:13662)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3240)
                                                                                        at android.view.View.getDisplayList(View.java:13536)
                                                                                        at android.view.View.getDisplayList(View.java:13662)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3240)
                                                                                        at android.view.View.getDisplayList(View.java:13536)
                                                                                        at android.view.View.getDisplayList(View.java:13662)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3240)
                                                                                        at android.view.View.getDisplayList(View.java:13536)
                                                                                        at android.view.View.getDisplayList(View.java:13662)
                                                                                        at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3240)
                                                                                        at android.view.View.getDisplayList(View.java:13536)
                                                                                        at android.view.View.getDisplayList(View.java:13662)
                                                                                        at android.view.HardwareRenderer$GlRenderer.buildDisplayList(HardwareRenderer.java:1666)
                                                                                        at android.view.HardwareRenderer$GlRenderer.draw(HardwareRenderer.java:1541)
                                                                                        at android.view.ViewRootImpl.draw(ViewRootImpl.java:2775)
                                                                                        at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2637)
                                                                                        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2253)
                                                                                        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1210)
                                                                                        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6408)
                                                                                        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:807)
                                                                                        at android.view.Choreographer.doCallbacks(Choreographer.java:610)
                                                                                        at android.view.Choreographer.doFrame(Choreographer.java:578)
                                                                                        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:793)
                                                                                        at android.os.Handler.handleCallback(Handler.java:808)
                                                                                        at android.os.Handler.dispatchMessage(Handler.java:103)
                                                                                        at android.os.Looper.loop(Looper.java:193)
                                                                                        at android.app.ActivityThread.main(ActivityThread.java:5521)
                                                                                        at java.lang.reflect.Method.invokeNative(Native Method)
                                                                                        at java.lang.reflect.Method.invoke(Method.java:515)
                                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:879)
                                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:695)
                                                                                        at dalvik.system.NativeStart.main(Native Method)
```

**Comment by ckenergy** [comment]
![screenshot_2017-12-28-15-09-02](https://user-images.githubusercontent.com/13145703/34403347-a39cb534-ebe2-11e7-8906-48f4ed4d2518.jpeg)

**Comment by diareuse** [comment]
I'm as well experiencing sudden increase after updating to 3.0.3. Every system version is affected with no difference of device brand.

@PhilJay is it possible to override LineChart and catch it in say `onDraw()` method?

**Comment by almic** [comment]
fixed now in d67ea481af1e8528c617dfd404d0c7827b0134b2
